tdms/include/input_output_names.h

```diff
@@ -0,0 +1,32 @@
+/**
+ * @file input_output_names.h
+ * @brief Constant values associated with the format of the input and output files and matrices
+ */
+#pragma once
+
+#define NMATRICES 49              //< number of input matrices
+#define NOUTMATRICES_WRITE 23     //< number of output matrices to be written to output file
+#define NOUTMATRICES_WRITE_ALL 25 //< number of output matrices to be written to output file
+#define NOUTMATRICES_PASSED 31    //< number of output matrices passed by mexFunction
+
+/*
+These are the names of the arrays that we expect to recieve from the input file (and posibly gridfile) that is passed to tdms on the command line.
+In the case where we are given an input file and a gridfile, the array names we expect to recieve from each is also recorded.
+In the case where the -m (compressed output) flag is passed, we do not save all the possible output arrays and so have a complete list of outputs and a shortened list of those which are saved when using -m.
+*/
+namespace tdms_matrix_names {
+    // all matrices that we could expect to get from an input file
+    extern const char *matrixnames[NMATRICES];
+    // matricies we expect to get from an input file if also provided a gridfile
+    extern const char *matrixnames_infile[NMATRICES-1];
+    // matrices we expect to obtain from a separate gridfile
+    extern const char *matrixnames_gridfile[1];
+    // oall output matrices we might want to write
+    extern const char *outputmatrices_all[NOUTMATRICES_WRITE_ALL];
+    // output matrices we want to write in a compressed (-m) output
+    extern const char *outputmatrices[NOUTMATRICES_WRITE];
+    // indices of the matrices to save when saving a complete system
+    extern int matricestosave_all[NOUTMATRICES_WRITE_ALL];
+    // indices of the matrices to save when saving a compressed output
+    extern int matricestosave[NOUTMATRICES_WRITE];
+}
```

Comment:
Ah, this is likely going to clash with my refactoring/reworking. (You weren't to know because I didn't tell you or push to my draft PR).

I went for `std::vector<std::string>` (which could be defined as const and live in a namespace). My reasoning: a vector knows its own `size_t` and one can do much nicer-looking loops like:
```c++
for (auto matrixname : matrixnames) {
     file.read(matrixname);
}
```

Reply:
Come to think of it, the matrices in our input and output files _define_ our file type. So perhaps a `tdms::File` is a HDF5 of this format.